torch/_inductor/fx_passes/post_grad.py

```diff
@@ -81,8 +82,27 @@ def visit(other_node):
             node.prepend(other_node)
 
     seen_nodes = set()
+
+    # only reorder nodes before the first copy_ in the graph.
+    # copy_ will appear at the end of functionalized graphs when there is mutation on inputs,
+    # and this reordering doesnt work well with mutation
+    first_copy = next(
+        (
+            node
+            for node in graph.nodes
+            if node.op == "call_function"
+            and node.target == torch.ops.aten.copy_.default
+        ),
+        None,
+    )
+
     for node in reversed(graph.nodes):
         seen_nodes.add(node)
+        if first_copy is not None:
```

Comment:
im missing where 'first_copy' causes a change in reordering behavior.  it looks like you just find the first copy here and continue adding all the nodes to the graph?

Reply:
I'm only doing reordering before the copy_ epilogue (only reordering before the first copy_ in the graph)